cmd/aro/monitor.go

```diff
@@ -126,7 +127,18 @@ func monitor(ctx context.Context, log *logrus.Entry) error {
 		return err
 	}
 
-	mon := pkgmonitor.NewMonitor(log.WithField("component", "monitor"), dialer, dbMonitors, dbOpenShiftClusters, dbSubscriptions, m, clusterm, liveConfig)
+	options := _env.Environment().EnvironmentCredentialOptions()
+	spTokenCredential, err := azidentity.NewEnvironmentCredential(options)
+	if err != nil {
+		return err
+	}
+
+	spGraphClient, err := _env.Environment().NewGraphServiceClient(spTokenCredential)
```

Comment:
Have we verified this will work?  

This environment credential option reads a service principal from environment variables, and we're not setting them on the monitor.  

Additionally, how do we plan on accomplishing this?  If we use the customer's cluster SP, it may not have the proper permissions to read its own expiry.

Reply:
The spGraphClient is passed from Monitor to Cluster package via pkg/monitor/worker.go.

Currently using msgraph-sdk-go we are fetching SP Id using APP Id in Graph Package. I created new function GetServicePrincipalExpiryByAppID using above logic under Graph package and made the API call to msgraph-sdk-go to fetch SP Expiry date using APP Id. If there is a Permission restriction it will impact both.

Reply:
I'm not concerned with the passing of the graph client, I'm concerned with how it's constructed.  

>This environment credential option reads a service principal from environment variables, and we're not setting them on the monitor. 

If you look at how the monitor is constructed from a systemd unit, we're not passing any [environment variables to the monitor process.](https://github.com/Azure/ARO-RP/blob/861c83319e8e7c59dab4011a8cc4c9f997509bc8/pkg/deploy/generator/scripts/rpVMSS.sh#L386-L423)

How are we testing this actually works as intended?  Additionally, if we use the first party service principal to check the expiration, the FPSP doesn't have the proper Microsoft.Graph permissions within the customer's tenant in order to check this expiration.